server/service/service_osquery.go

```diff
@@ -336,10 +346,18 @@ func (svc *Service) GetClientConfig(ctx context.Context) (map[string]interface{}
 	}
 
 	if saveHost {
-		err := svc.ds.SaveHost(ctx, &host)
-		if err != nil {
-			return nil, err
-		}
+		go func() {
+			atomic.AddInt64(&counter, 1)
```

Comment:
To avoid another `atomic` call on line 354, you can store the returned value here and use it in the debug call:
```suggestion
			newVal := atomic.AddInt64(&counter, 1)
```